### PR TITLE
Add repository list page + it's related modals (PROJQUAY-3892)

### PR DESCRIPTION
**Pull request by harishsurf**
This task covers:

- [PROJQUAY-3892](https://issues.redhat.com/browse/PROJQUAY-3892)
- [PROJQUAY-3900](https://issues.redhat.com/browse/PROJQUAY-3900)
- [PROJQUAY-3896](https://issues.redhat.com/browse/PROJQUAY-3896)

Changes include: 
- Repository List page + Create repository modal
- Modals for changing repo visibility to `private/public` and deleting repositories
- Repositories tab under OrganizationList page -> Specific org -> Repositories tab
- Add navigation for all pages

ToDo:
- Allow multiple repo deletion
- Update list results after changing visibility (presently you need to refresh the page)

Signed-off-by: harishsurf <hgovinda@redhat.com>

### CODE REVIEW

**Comment by syed** [comment]
from last meeting: move the generic repositories component to the global components directory

File: src/components/modals/CreateRepoModalTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+  //   ));
+  //   return [
+  //     ...children,
+  //     <Divider component="li" />,
+  //     <SelectOption>{loggedInUser} </SelectOption>,
+  //   ];
+  // };
+
+  const createRepositoryHandler = async () => {
+    handleModalToggle(); // check if this is needed
```
Comment (Sunandadadi) [inline src/components/modals/CreateRepoModalTemplate.tsx:74]: Do you want to remove the comment if you've checked this?

File: src/components/modals/CreateRepoModalTemplate.tsx [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+  const handleNameInputChange = (value) => {
+    setNewRepository({...newRepository, name: value});
+  };
+
+  const handleRepoDescriptionChange = (value) => {
+    setNewRepository({...newRepository, description: value});
+  };
+
+  // const orgsDropdown = () => {
```
[the remark this answers is not in the document]
Comment (harishsurf) [reply src/components/modals/CreateRepoModalTemplate.tsx:62]: replaced it with a TODO action for myself

File: src/components/modals/CreateRepoModalTemplate.tsx (hunk 1 above)
Comment (harishsurf) [reply src/components/modals/CreateRepoModalTemplate.tsx:74]: turns out, it's needed

**Review by Sunandadadi — APPROVED** [review]

**Pull request merged**